ui/StatusQ/include/StatusQ/aggregator.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <QAbstractListModel>
+#include <QObject>
+#include <QVariant>
+
+class Aggregator : public QObject {
+    Q_OBJECT
+
+    Q_PROPERTY(QAbstractItemModel* model READ model WRITE setModel NOTIFY modelChanged)
+    Q_PROPERTY(QString roleName READ roleName WRITE setRoleName NOTIFY roleNameChanged)
+    Q_PROPERTY(QVariant value READ value NOTIFY valueChanged)
+
+public:
+    explicit Aggregator(QObject *parent = nullptr);
+
+    QAbstractItemModel* model() const;
+    void setModel(QAbstractItemModel* model);
+
+    QString roleName() const;
+    void setRoleName(const QString &roleName);
```

Comment:
Ok.. I get the point and I can partially agree, however, if we exclude the role name management in the abstract base `Aggregator` class.. what this would pretend to do?? Just listening to the model changes and provide a virtual method to overide??  

Another alternative, that will make the `Aggregator` implementation a bit more complex (but I guess not so much) is to change `roleName` type to be a `QList<QByteArray>` so we can manage more than one role name and we don't need to create another specific abstract class for single roles. WDYT?

Reply:
Added `SingleRoleAggregator`!